cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -4165,10 +4165,11 @@ void TryReadDataFile(const std::string& path,
   auto pool = ::arrow::default_memory_pool();
 
   std::unique_ptr<FileReader> arrow_reader;
-  Status s =
-      FileReader::Make(pool, ParquetFileReader::OpenFile(path, false), &arrow_reader);
-  if (s.ok()) {
+  Status s;
+  Result result = FileReader::Make(pool, ParquetFileReader::OpenFile(path, false));
+  if (result.ok()) {
```

Comment:
```suggestion
  auto reader_result = FileReader::Make(pool, ParquetFileReader::OpenFile(path, false));
  if (reader_result.ok()) {
```

(I like `XXX_result` but others use `maybe_XXX` or something.)